services/core-api/app/api/mines/documents/resources/mine_document_resource.py

```diff
@@ -127,6 +134,12 @@ def patch(self, mine_guid):
 
         MineDocument.mark_as_archived_many(mine_document_guids)
 
+        if len(mine_document_guids) > 0:
+            project = ProjectsSearchUtil.find_by_mine_document_guid(mine_document_guids[0])
```

Comment:
I think this is still too specific to projects.

Maybe pass in a parameter to represent the "activity type" and do something like a switch (ie: if parentEntityType == project then <do project related things>), or make a separate call to do the notification.

Reply:
@taraepp 
This parameter `activityType` you suggest, it should be passed from the front end?

Reply:
@taraepp @isuru-aot I would actually suggest not to pass along other parameters here. In the end I think it's the backend that should decide what kind of document it is - we don't accidentally want to send out notifications for other types on documents

Reply:
Updated as discussed to decide the doc is valid to send notification.